### DISCUSSION TITLE
turborepo does not run pipeline dependsOn task

### DISCUSSION THREAD

**Discussion by lovetingyuan** [vercel/turborepo#849] [Help]
```json
{
  "playground#dev": {
    "cache": false,
    "outputs": ["dist/**"],
    "dependsOn": ["pages#build", "^types"]
  },
  "dev": {
    "cache": false,
    "outputs": ["dist/**"]
  },
  "types": {
    "outputs": ["dist/**"],
    "cache": true,
    "dependsOn": ["^types"]
  }
}
```
I run turbo run dev, playground dev script will be executed, before this, I expect build script of pages and every `types` command of dependencies will be  executed, but the fact is only build script of pages is executed, while the `types` script is ignored.

Could someone tell me the reason? Is the configuration wrong?

**Comment by lovetingyuan** [comment]
I modify source code, I expect `types` script will be executed, but turborepo does not run "types" script.

**Comment by lovetingyuan** [comment]
```json
"playground#dev": {
      "cache": false,
      "outputs": [
        "dist/**"
      ],
      "dependsOn": ["pages#build", "another#types"]
    },
    "dev": {
      "cache": false,
      "outputs": [
        "dist/**"
      ]
    },
    "types": {
      "outputs": [
        "dist/**"
      ],
      "cache": true,
      "dependsOn": ["^types"]
    }

```

If I use this config, both "pages#build" and "another#types" will be executed.

**Comment by lovetingyuan** [comment]
who can help me?

**Comment by denyskotsur** [comment]
@lovetingyuan  any updates on this? Have you figured out this?

**Comment by mehulkar** [comment]
@lovetingyuan if `playground#dev` is a long running task (e.g. a server), it will not re-run its dependencies while it's running. You are probably looking for a Watch Mode feature which Turborepo currently does not have https://github.com/vercel/turbo/issues/986.
